contrib/meson/meson.build

```diff
@@ -40,6 +40,9 @@ libzstd_includes = [include_directories(common_dir, dictbuilder_dir, compress_di
 
 legacy = get_option('legacy_support')
 if legacy != 'false'
+    if legacy == 'true'
+        legacy = '0'
```

Comment:
Could you describe what this statement is expected to achieve ?

Reply:
it allos you to do "meson configure -Dlegacy=true" and have it equivalent to make -DZSTD_LEGACY_SUPPORT=0 (which compiles in all legacy options). That 0 does that is not so clear.

Reply:
Not that I see the documentation I see this is totally wrong, I will update the patch

Reply:
You probably want a positive value >= 1 here